superset-frontend/src/SqlLab/components/EstimateQueryCostButton/index.tsx

```diff
@@ -26,24 +25,28 @@ import Loading from 'src/components/Loading';
 import ModalTrigger from 'src/components/ModalTrigger';
 import { EmptyWrapperType } from 'src/components/TableView/TableView';
 
-const propTypes = {
-  dbId: PropTypes.number.isRequired,
-  schema: PropTypes.string.isRequired,
-  sql: PropTypes.string.isRequired,
-  getEstimate: PropTypes.func.isRequired,
-  queryCostEstimate: PropTypes.Object,
-  selectedText: PropTypes.string,
-  tooltip: PropTypes.string,
-  disabled: PropTypes.bool,
-};
-const defaultProps = {
-  queryCostEstimate: [],
-  tooltip: '',
-  disabled: false,
-};
+interface EstimateQueryCostButtonProps {
+  dbId: number;
+  schema: string;
+  sql: string;
+  getEstimate: Function;
+  queryCostEstimate: Record<string, any>;
```

Comment:
You're saying this is a record, but using an empty array as the default value. I know that's what was used before, but something probably needs to change here.

Reply:
Just for the sanity check, I narrowed down on this further and found that the `SqlEditor` component makes use of the `EstimateQueryCostButton` component with the `queryCostEstimate={qe.queryCostEstimate}` prop set. Interestingly, the type for the `qe.queryCostEstimate` argument being passed here is inferred to be `any`.